File: LibraryProject_C++/Media.cpp
#include "Media.h"


Media::Media(const string& call_no,const string& title,const string& subject,const string& notes):CallNo(call_no),Title(title),Subject(subject),Notes(notes){
	
}

Media::~Media(){}


/*
   compare_Callno returns true if call number is found.
Input: Constant String reference
Output:boolean value
 */
bool Media::compare_Callno(const string& call_no) const{

	if(CallNo.find(call_no)!=string::npos){  //string find returns npos if nothing is found
		return true;
	}
	else{
		return false;
	}
	return 1;
}

/*
   compare_Title returns true if requested title is found.
Input: Constant String reference
Output:boolean value
 */

bool Media::compare_Title(const string& title) const{

	if(Title.find(title)!=string::npos){
		return true;
	}
	else{
		return false;
	}
}

/*
   compare_Subject returns true if requested subject is found.
Input: Constant String reference
Output:boolean value
 */

bool Media::compare_Subject(const string& subject) const{

	if(Subject.find(subject)!=string::npos){
		return true;
	}
	else{
		return false;
	}
}

/*
   compare_other returns true if any random string passed to it is found.
Input: Constant String reference
Output:boolean value
 */

bool Media::compare_other(const string& other) const{

	if(Notes.find(other)!=string::npos){
		return true;
	}
	else{
		return false;
	}
}

/*
   display function prints only the four common fields for book, periodics, video and film.
 */
void Media::display() const{

	cout<<"Call Number: "<<CallNo<<endl;
	cout<<"Title:       "<<Title<<endl;
	cout<<"Subject:     "<<Subject<<endl;
	cout<<"Notes:       "<<Notes<<endl;   
}

File: libraryProject_Python/Media.py
#Author: Akshay S. Bembalkar

class Media(object):

#Constructor to initialize attributes those are common to book, film, videos and periodic.
	def __init__(self,callNo,title,subject,notes):
		self.callNo = callNo
		self.title = title
		self.subject = subject
		self.notes = notes



#Comapare string with call number
	def compare_Callno(self,callNo):
		if callNo in self.callNo:     #If string is in Call number string,
			return True           #Return true
		else:
			return False


#Comapare string with Title
	def compare_Title(self,title):
		if title in self.title:
			return True
		else:
			return False
	
#Comapare string with Subject
	def compare_Subject(self,subject):
		if subject in self.subject:
			return True
		else:
			return False

	def display(self):
		print ("Call Number: %s" % (self.callNo))
		print ("Title: %s" % (self.title))
		print ("Subject: %s" % (self.subject))
		print ("Notes: %s" % (self.notes))
